exec/java-exec/src/main/java/org/apache/drill/exec/coord/zk/ZKClusterCoordinator.java

```diff
@@ -231,31 +235,42 @@ public DrillbitEndpoint apply(ServiceInstance<DrillbitEndpoint> input) {
       Set<DrillbitEndpoint> unregisteredBits = new HashSet<>(endpoints);
       unregisteredBits.removeAll(newDrillbitSet);
 
+      // Set of newly live bits : new set of active bits - original bits.
+      Set<DrillbitEndpoint> registeredBits = new HashSet<>(newDrillbitSet);
+      registeredBits.removeAll(endpoints);
+
       endpoints = newDrillbitSet;
 
       if (logger.isDebugEnabled()) {
         StringBuilder builder = new StringBuilder();
         builder.append("Active drillbit set changed.  Now includes ");
         builder.append(newDrillbitSet.size());
-        builder.append(" total bits.  New active drillbits: \n");
-        for (DrillbitEndpoint bit: newDrillbitSet) {
-          builder.append('\t');
-          builder.append(bit.getAddress());
-          builder.append(':');
-          builder.append(bit.getUserPort());
-          builder.append(':');
-          builder.append(bit.getControlPort());
-          builder.append(':');
-          builder.append(bit.getDataPort());
-          builder.append('\n');
+        builder.append(" total bits.");
+        if (!newDrillbitSet.isEmpty()) {
+          builder.append(" New active drillbits: \n");
+          for (DrillbitEndpoint bit: newDrillbitSet) {
+            builder.append('\t');
+            builder.append(bit.getAddress());
+            builder.append(':');
+            builder.append(bit.getUserPort());
+            builder.append(':');
+            builder.append(bit.getControlPort());
+            builder.append(':');
+            builder.append(bit.getDataPort());
+            builder.append('\n');
+          }
+          logger.debug(builder.toString());
         }
-        logger.debug(builder.toString());
       }
 
-      // Notify the drillbit listener for newly unregistered bits. For now, we only care when drillbits are down / unregistered.
-      if (! (unregisteredBits.isEmpty()) ) {
+      // Notify the drillbit listener for newly unregistered bits.
+      if (!(unregisteredBits.isEmpty())) {
```

Comment:
Correct the comment to "Notify **all listeners** for newly unregistered bits." (similarly correction below).